code/modules/mob/living/bot/bot.dm

```diff
@@ -302,7 +302,9 @@
 /mob/living/bot/proc/startPatrol()
 	var/turf/T = getPatrolTurf()
 	if(T)
-		patrol_path = AStar(get_turf(loc), T, /turf/proc/CardinalTurfsWithAccess, /turf/proc/Distance, 0, max_patrol_dist, id = botcard, exclude = obstacle)
+		busy = 1	// Multi-z can take a bit. This'll make sure it isn't called twice while it runs
+		patrol_path = AStarWithZ(get_turf(loc), T, /turf/proc/CardinalTurfsWithAccess, /turf/proc/Distance, 0, max_patrol_dist, id = botcard, exclude = obstacle)
```

Comment:
To illustrate the more elegant approach (mockup, but the distance procs should work):

```suggestion
/turf/proc/3dCardinalTurfsWithAccess(id_card)
    return all_adjacent_turfs_in_3d // somehow
    
/turf/proc/Euclidean3dDistance(turf/endloc)
   // equivalent to the current Distance proc metric, except for 3D
   var/euclid_dist =(src.x - t.x) * (src.x - t.x) + (src.y - t.y) * (src.y - t.y) + (src.z - t.z) * (src.z - t.z)
   euclid_dist *= (pathweight+t.pathweight)/2
   return euclid_dist
    
/turf/proc/Manhattan3dDistance(turf/endloc)
   // just to show off that we can use multiple metrics; AKA taxicab distance
   var/manhattan_dist =abs((src.x - t.x)) + abs((src.y - t.y)) + abs((src.z - t.z))
   manhattan_dist *= (pathweight+t.pathweight)/2
   return manhattan_dist
   
		patrol_path = AStar(get_turf(loc), T, /turf/proc/3dCardinalTurfsWithAccess, /turf/proc/Euclidean3dDistance, 0, max_patrol_dist, id = botcard, exclude = obstacle)
```

Replace the procs like this for anything you want to be 3D-capable.